### PR TITLE
[RETINA-347] Improve duplicate sn mn message

### PR DESCRIPTION
**Pull request by Jbcampbe**
In response to #347 

- Add checks for network error on api calls
- Add rollbar error logging
- Add checks for a few specific api status codes and display better error messages

Additional fixes
- Configuration page button would get moved by long text
- user's names were to close to their initials on the user summary page

### CODE REVIEW

File: src/routes/configuration.vue [hunk 1]
```diff
@@ -215,22 +222,35 @@ export default {
...
         this.$apollo.queries.getAllConfigurableItem.refetch()
-      }).catch(response => {
-        this.showInvalidItemMsg()
+      }).catch(error => {
+        if (handleCommonErrors(error)) {
+          return
+        }
+
+        if (error && error.graphQLErrors && error.graphQLErrors.length && error.graphQLErrors[0].extensions.code === ApiStatusCodes.CONFIGURABLE_ITEM_UNIQUE_CONSTRAINT_VIOLATION) {
+          this.showInvalidItemMsg()
```
Comment (ekraihan) [inline src/routes/configuration.vue]: Will this message and the one caused by showErrorMsg() be shown? Will that look bad?

File: src/store/modules/tools.js [hunk 2]
```diff
@@ -132,8 +142,16 @@ const tools = {
 
         return response
       } catch (error) {
-        window.console.error(error)
-        showErrorMsg()
+        if (handleCommonErrors(error)) {
+          return
+        }
+
+        if (error && error.graphQLErrors && error.graphQLErrors.length && error.graphQLErrors[0].extensions.code === ApiStatusCodes.TOOL_UNIQUE_IN_SERVICE_CONSTRAINT_VIOLATION) {
```
Comment (ekraihan) [inline src/store/modules/tools.js]: possibly wrap `error && error.graphQLErrors && error.graphQLErrors.length && error.graphQLErrors[0].extensions.code` in a function? like
```javascript
function getErrorCode(error) {
    if (error && error.graphQLErrors && error.graphQLErrors.length) {
        return error.graphQLErrors[0].extensions.code
    }
}
```

**Comment by ekraihan** [comment]
When I sign in with incorrect credentials, I get this message and then I get a thinking spinner that spins indefinitely
image

File: src/routes/configuration.vue (hunk 1 above)
Comment (Jbcampbe) [reply src/routes/configuration.vue]: good catch. it won't look bad but the user won't have time to read the invalid item message before it gets covered up by the generic message. fixed

**Pull request merged**